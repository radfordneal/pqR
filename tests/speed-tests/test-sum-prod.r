source("time.r")
 
test.name <- "sum-prod.sum"

a <- seq(0.001,1,length=1000)
a3 <- c(1,2,3)

f <- test.cmp (function () {for (i in 1:5000000) b <- sum(a3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:500000) b <- sum(a); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:500000) b <- sum(a,na.rm=TRUE); b})
sys.time(b<-f()); print(b)

a[100] <- NA
f <- test.cmp (function () {for (i in 1:500000) b <- sum(a,na.rm=TRUE); b})
sys.time(b<-f()); print(b)

x <- 1:1000

f <- test.cmp (function () {for (i in 1:500000) b <- sum(x); b})
sys.time(b<-f()); print(b)

z <- seq(0.001,1,length=1000) + 1i

f <- test.cmp (function () {for (i in 1:500000) b <- sum(z); b})
sys.time(b<-f()); print(b)

test.name <- "sum-prod.sum-math1"

a <- seq(0.001,1,length=1000)
aa <- seq(0.001,1,length=10000)

f <- test.cmp (function () {for (i in 1:50000) b <- sum(log(a)); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:50000) b <- sum(exp(a)); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:100000) b <- sum(trunc(a)); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:500000) b <- sum(abs(a)); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:10000) b <- sum(sqrt(aa)); b})
sys.time(b<-f()); print(b)

test.name <- "sum-prod.prod"

a <- seq(0.9,1.1,length=1000)
a3 <- c(1,2,3)

f <- test.cmp (function () {for (i in 1:5000000) b <- prod(a3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:500000) b <- prod(a); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:500000) b <- prod(a,na.rm=TRUE); b})
sys.time(b<-f()); print(b)

a[100] <- NA
f <- test.cmp (function () {for (i in 1:500000) b <- prod(a,na.rm=TRUE); b})
sys.time(b<-f()); print(b)

x <- rep(1:2,500)

f <- test.cmp (function () {for (i in 1:500000) b <- sum(x); b})
sys.time(b<-f()); print(b)

z <- seq(0.9,1.1,length=1000) + 1i

f <- test.cmp (function () {for (i in 1:500000) b <- sum(z); b})
sys.time(b<-f()); print(b)
