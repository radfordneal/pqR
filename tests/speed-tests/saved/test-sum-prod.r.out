
---- sum-prod.sum 

Time:
[1] 6
Time:
[1] 500.5
Time:
[1] 500.5
Time:
[1] 500.4
Time:
[1] 500500
Time:
[1] 500.5+1000i

---- sum-prod.sum-math1 

Time:
[1] -995.6271
Time:
[1] 1719.141
Time:
[1] 1
Time:
[1] 500.5
Time:
[1] 6672.977

---- sum-prod.prod 

Time:
[1] 6
Time:
[1] 0.1872993
Time:
[1] 0.1872993
Time:
[1] 0.2036261
Time:
[1] 1500
Time:
[1] 1000+1000i
